### PR TITLE
Accept empty dataframes in DataFrame.to_parquet

### LINKED ISSUE #27339: DataFrame.to_parquet raises fails when dataframe is empty
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

df = pd.DataFrame()
pd.to_parquet('empty.parquet')
```

#### Problem description

`DataFrame.to_parquet` raises `ValueError: parquet must have string column names` when the dataframe has no columns. `pyarrow` and `fastparquet` both support writing empty dataframes so I don't think this restriction is necessary (see below script).

```py
import fastparquet as fp
import pyarrow as pa
import pyarrow.parquet as pq
import pandas as pd

df = pd.DataFrame()
fn = 'empty.parquet'

# Test fastparquet
fp.write(fn, df)

# Test pyarrow
table = pa.Table.from_pandas(df)
pq.write_table(table, fn)

# This will fail
df.to_parquet(fn)
```

I'm happy to open a PR if it's agreed that this is an issue.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Darwin
OS-release: 18.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: None
pip: 19.1.1
setuptools: 41.0.1
Cython: None
numpy: 1.16.4
scipy: None
pyarrow: 0.14.0
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.8.0
pytz: 2019.1
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by CJStadler**
Fixes #27339

I wrote two tests because `fastparquet` adds a name to the index when it deserializes.

- [x] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -471,6 +471,10 @@ def test_partition_cols_supported(self, pa, df_full):
             assert len(dataset.partitions.partition_names) == 2
             assert dataset.partitions.partition_names == set(partition_cols)
 
+    def test_empty_dataframe(self, pa):
+        df = pd.DataFrame()
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: can you add the github issue number as a comment

File: pandas/tests/io/test_parquet.py [hunk 2]
```diff
@@ -566,3 +570,9 @@ def test_error_on_using_partition_cols_and_partition_on(self, fp, df_full):
                     partition_on=partition_cols,
                     partition_cols=partition_cols,
                 )
+
+    def test_empty_dataframe(self, fp):
+        df = pd.DataFrame()
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: same here

**Review by jreback — CHANGES_REQUESTED** [review]
small comments, can you add a whatsnew note (bug fixes in io), ping on green.

**Comment by CJStadler** [comment]
Thanks @jreback. Should this go in the whatsnew for 0.25.0 or 0.25.1?

**Comment by jreback** [comment]
> Thanks @jreback. Should this go in the whatsnew for 0.25.0 or 0.25.1?

you can put in 0.25.0

**Review by jorisvandenbossche — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @CJStadler ping on green.

**Comment by CJStadler** [comment]
It's green @jreback. Thanks!

**Comment by jreback** [comment]
thanks!

**Pull request merged**